packages/flutter/lib/src/material/input.dart

```diff
@@ -137,6 +147,15 @@ class _InputState extends State<Input> {
       config.onSubmitted(_value);
   }
 
+  void _handleSelectionChanged(TextSelection selection) {
+    if (_isAttachedToKeyboard) {
+      _keyboardHandle.setSelection(selection.start, selection.end);
```

Comment:
Yeah, we need to reorganize this code a bit.  There's another problem where the EditableString does't know how to tell the render object to repaint when the selection changes.  Currently we wait for the cursor to blink.  I can look at this in the next patch.